pkg/controllers/workspace_controller.go

```diff
@@ -287,11 +289,29 @@ func (c *WorkspaceReconciler) updateControllerRevision(ctx context.Context, wObj
 
 			if hash, exists := deployment.Annotations[WorkspaceRevisionAnnotation]; !exists || (hash != currentHash) {
 
-				initContainers, envs := resources.GenerateInitContainers(wObj)
+				var volumes []corev1.Volume
+				var volumeMounts []corev1.VolumeMount
+				shmVolume, shmVolumeMount := utils.ConfigSHMVolume(*wObj.Resource.Count)
+				if shmVolume.Name != "" {
+					volumes = append(volumes, shmVolume)
+				}
+				if shmVolumeMount.Name != "" {
+					volumeMounts = append(volumeMounts, shmVolumeMount)
+				}
+
+				if len(wObj.Inference.Adapters) > 0 {
```

Comment:
Ok, I see you also want to support a case for adapters are being removed from the spec, in this case, the initcontainer sure just be empty.  Can you adjust the code to achieve that?